robot/impl/local_robot.go

```diff
@@ -646,6 +646,7 @@ func (r *localRobot) getOptionalDependencies(conf resource.Config) resource.Depe
 				"resource", conf.ResourceName().String(),
 				"error", err,
 			)
+			continue
```

Comment:
Caught a bug with my new test. We need to `continue` when the optional dependency for the resource exists but is not available so that the optional dependencies name is not passed down to the resource's `Reconfigure` method (or constructor).